web_app_drawer/static/src/js/web_app_drawer.js

```diff
@@ -0,0 +1,232 @@
+            this.$el.one('drawer.opened', $.proxy(this.onDrawerOpen, this));
+        },
+        
+        // Generate x and y coords for app icons
+        onDrawerOpen: function() {
+            this.isOpen = true;
+            this.$appLinks = $('.app-drawer-icon-app').parent();
+            this.selectAppLink($(this.$appLinks[0]));
+            this.$appLinks.on('mouseenter', $.proxy(this.selectAppLinkEvent, this));
+            this.$el.one('drawer.closed', $.proxy(this.onDrawerClose, this));
+            core.bus.trigger('drawer.opened');
+        },
+        
+        // It provides a proxy method allowing for app link select w/ events
+        selectAppLinkEvent: function(event) {
+            this.selectAppLink($(event.currentTarget));
+        },
+        
+        // It selects an app link visibly
+        selectAppLink: function($appLink) {
+            if ($appLink) {
+                this.$selectedAppLink = $appLink;
+                $appLink.focus();
+            }
+        },
+        
+        /* It searches for app links by name and returns the first match
+         * @param query str to search
+         * @return jQuery obj
+         */
+        searchAppLinks: function(query) {
+            return this.$appLinks.filter(function() {
+                return $(this).data('menuName').toUpperCase().startsWith(query);
+            }).first();
+        },
+        
+        /* Find the link adjacent to $appLink in provided direction
+         * @param $appLink jQuery obj of App icon link
+         * @param direction Array for direction to look [int:x, int:y]
+         * @return jQuery obj for adjacent applink
+         */
+        findAdjacentAppLink: function($appLink, direction) {
+            
+            var obj = [],
+                $objs = this.$appLinks;
+            
+            switch(direction){
+                case this.LEFT:
+                    obj = $objs[$objs.index($appLink) - 1];
+                    if (!obj) {
+                        obj = $objs[$objs.length - 1];
+                    }
+                    break;
+                case this.RIGHT:
+                    obj = $objs[$objs.index($appLink) + 1];
+                    if (!obj) {
+                        obj = $objs[0];
+                    }
+                    break;
+                case this.UP:
+                    $objs = this.getRowObjs($appLink, this.$appLinks);
+                    obj = $objs[$objs.index($appLink) - 1];
+                    if (!obj) {
+                        obj = $objs[$objs.length - 1];
+                    }
+                    break;
+                case this.DOWN:
+                    $objs = this.getRowObjs($appLink, this.$appLinks);
+                    obj = $objs[$objs.index($appLink) + 1];
+                    if (!obj) {
+                        obj = $objs[0];
+                    }
+                    break;
+            }
+            
+            return $(obj);
```

Comment:
Please add `if (obj.length) {event.preventDefault()}` to avoid background views scrolling when pressing U/D arrows.